src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,53 @@
+package duke.command;
+
+import duke.exception.DukeException;
+
+/**
+ * Parser parses the input by the user and determines what the command is the user is issuing, or no command at all
+ * (invalid command).
+ */
+public class Parser {
+
+    /**
+     * Returns the correct command object, or an exception if the user input is invalid in format.
+     * @param fullCommand The full string input of the user
+     * @return An object that is a child class of Command
+     * @throws DukeException DukeException is thrown if the command is not valid
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        assert !fullCommand.equals(""): "The command is empty.";
+        fullCommand = fullCommand.trim();
+        String[] commandWords = fullCommand.split("\\s+");
+        String commandType = commandWords[0];
+        if (commandType.equals("todo")) {
```

Comment:
This entire block might look neater if handled by a `switch` block instead. Each case could also be abstracted out.